src/wasm.h

```diff
@@ -71,6 +71,7 @@ struct Name : public cashew::IString {
   Name() : cashew::IString() {}
   Name(const char *str) : cashew::IString(str) {}
   Name(cashew::IString str) : cashew::IString(str) {}
+  Name(const std::string &str) : cashew::IString(str.c_str()) {}
```

Comment:
Yes. The idea was that those are coming from a buffer, similar to the original use case. But agreed, that's not good. I'll change that.